### PR TITLE
docs: Update version matrix

### PR DESCRIPTION
**Pull request by sayboras**
This is to address the below comment.

https://github.com/cilium/cilium-cli/pull/2285#issuecomment-1926303693

### CODE REVIEW

**Comment by sayboras** [comment]
CI failure is not related for this doc change.

File: README.md [hunk 1]
```diff
@@ -43,7 +43,8 @@ binary releases.
 
 | Release                                                                | Maintained | Compatible Cilium Versions |
 |------------------------------------------------------------------------|------------|----------------------------|
-| [v0.15.23](https://github.com/cilium/cilium-cli/releases/tag/v0.15.23) | Yes        | Cilium 1.14 and newer      |
+| [v0.15.23](https://github.com/cilium/cilium-cli/releases/tag/v0.15.23) | Yes        | Cilium 1.15 and newer      |
+| [v0.15.22](https://github.com/cilium/cilium-cli/releases/tag/v0.15.22) | Yes        | Cilium 1.14                |
 | [v0.14.8](https://github.com/cilium/cilium-cli/releases/tag/v0.14.8)   | Yes        | Cilium 1.13                |
```
Comment (michi-covalent) [inline README.md:49]: could you maybe add a sentence or two here and describe why `>=v0.15.23` is incompatible with cilium v1.14?

Comment (sayboras) [reply README.md:49]: here is the reason https://github.com/cilium/cilium-cli/pull/2285#issuecomment-1926303693.

Do you want to mention in the docs as well ?

Comment (michi-covalent) [reply README.md:49]: yeah i think it makes sense to add it in README.md

Comment (sayboras) [reply README.md:49]: ah right, I have added a small footnote for v0.15.22.

**Review by michi-covalent — APPROVED** [review]

**Review by tklauser — COMMENTED** [review]
@sayboras this change is a bit unfortunate because we try to support all currently supported stable releases on the latest cilium-cli release. If I read the original change https://github.com/cilium/cilium-cli/pull/2285#issuecomment-1926303693 correctly, couldn't we instead just also capture `v1beta1` version of the changed resources in addition to the `v1` ones so we could at least still support Cilium 1.14 and Cilium 1.15 in the latest cilium-cli version.

**Comment by sayboras** [comment]
> couldn't we instead just also capture v1beta1 version of the changed resources in addition to the v1 ones so we could at least still support Cilium 1.14 and Cilium 1.15 in the latest cilium-cli version.

Yes, we can. There might be the case that we missed some attributes (i.e. available in v1, but not in v1beta1) in the future.

**Pull request merged**